ucte/ucte-network/src/main/java/com/powsybl/ucte/network/UcteValidation.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.ucte.network;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Anne Tilloy {@literal <anne.tilloy at rte-france.com>}
+ */
+public final class UcteValidation {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UcteValidation.class);
+
+    public static final double ZERO_EPS = 1e-4;
+    public static final double REACTANCE_EPS = 0.05;
+
+    private UcteValidation() {
+    }
+
+    // Data: line characteristics
+    public static void checkValidLineCharacteristics(UcteLine line) {
+        switch (line.getStatus()) {
+            case EQUIVALENT_ELEMENT_IN_OPERATION:
+            case EQUIVALENT_ELEMENT_OUT_OF_OPERATION:
+            case REAL_ELEMENT_IN_OPERATION:
+            case REAL_ELEMENT_OUT_OF_OPERATION:
+                if (line.getResistance() < ZERO_EPS) {
+                    LOGGER.error("Real line resistance cannot be negative");
+                }
+                if (Math.abs(line.getReactance()) > ZERO_EPS) {
+                    LOGGER.warn("Busbar coupler reactance must be zero. For lines this value must be larger than 0.05 ohm");
+                }
+                break;
+            case BUSBAR_COUPLER_IN_OPERATION:
+            case BUSBAR_COUPLER_OUT_OF_OPERATION:
+                if (Math.abs(line.getResistance()) > ZERO_EPS) {
+                    LOGGER.warn("Busbar coupler resistance must be zero");
+                }
+                if (Math.abs(line.getReactance()) > ZERO_EPS) {
+                    LOGGER.warn("Busbar coupler reactance must be zero");
+                }
+                if (Math.abs(line.getSusceptance()) > ZERO_EPS) {
+                    LOGGER.warn("Busbar coupler susceptance must be zero");
+                }
+            default:
```

Comment:
You have to do something in the default: in that specific case, throw an assertion error because we don't expect another line.status value.